SilKit/source/config/YamlConversion.cpp

```diff
@@ -124,7 +125,50 @@ bool Converter::decode(const Node& node, Sink& obj)
     return true;
 }
 
+
+template <>
+Node Converter::encode(const Sink::Format& obj)
+{
+    Node node;
+    switch (obj)
+    {
+    case Sink::Format::Simple:
+        node = "Simple";
+        break;
+    case Sink::Format::Json:
+        node = "Json";
+        break;
+    default:
```

Comment:
Usually a `default` case in a `switch` statement is a good idea, but in certain circumstances it can be problematic. If the `switch` is supposed to be exhaustive, the `default` case suppresses the compilers ability to emit a warning if the `enum` is extended without adding a new case to the `switch` statement.

The `default` statement does make sense for exhaustive `switch`-es if the acutal `enum` value comes from "user code" (i.e., code not under our control) - because `C++` does not mandate that the underlying integer is any of the `enum` values.

Reply:
I'm not 100% sure if we have setup our compiler warnings properly, so you can keep it this way if you like 👍 Just thought I'd mention it.